### PR TITLE
Avoid publishing partially filed out header

### PR DESCRIPTION
**Pull request by alamar**
see ChronicleEnterprise/Chronicle-Queue-Enterprise#415

### CODE REVIEW

**Review by nicktindall — APPROVED** [review]
This looks good to me, but @peter-k-lawrey has the final say

**Review by JerryShea — APPROVED** [review]
This is excellent. Would prefer we get @peter-lawrey tick of approval before merging though

**Pull request merged**
